Nos/Models/CoreData/Event+CoreDataClass.swift

```diff
@@ -826,18 +835,31 @@ public class Event: NosManagedObject, VerifiableEvent {
         guard !loadingViewData else {
             return
         }
-        loadingViewData = true 
+        
+        loadingViewData = true
         Log.debug("\(identifier ?? "null") loading view data")
         
-        if isStub {
-            await loadContent()
-            loadingViewData = false
-            // TODO: how do we load details for the event again after we hydrate the stub?
-        } else {
-            Task { await loadReferencedNote() }
-            Task { await loadAuthorMetadata() }
-            Task { await loadAttributedContent() }
+        await withTaskGroup(of: Void.self) { group in
+            if isStub {
+                group.addTask {
+                    await self.loadContent()
+                }
+                // TODO: how do we load details for the event again after we hydrate the stub?
+            } else {
+                group.addTask {
+                    await self.loadReferencedNote()
+                }
+                group.addTask {
+                    await self.loadAuthorMetadata()
+                }
+                group.addTask {
+                    await self.loadAttributedContent()
+                }
+            }
+            
+            await group.waitForAll()
         }
+        loadingViewData = false
```

Comment:
This line of code was only called in the else condition before, I just want to double check having it here is desirable.

Reply:
I believe that it is desirable. Otherwise, `loadingViewData` will never get set back to `false`, which could be misleading both to UI that depends on it as well as to devs trying to debug issues in this area. Would be great to get confirmation from @mplorentz if possible though.